release-engineer-tools/prepare-signed-flasher-bundle.sh

```diff
@@ -0,0 +1,204 @@
+#!/bin/bash
+#
+# ------------------------------------------------------------------------------
+# This script is meant to automate and guard rail the work of a release engineer
+# to take an unsigned-flasher-bundle produced by the "prod.secure" build plan
+# and make it into signed-flasher-bundle that is then used by:
+# - production systems in the factory
+# - field customers to perform OTA Software Update
+#
+# The high level steps are:
+#
+# 1. Extract and validate the input
+# 2. Sign binaries used for production systems
+# 3. Generate encrypted ota for field customer upgrade
+# 4. Re-archive the updated package
+# ------------------------------------------------------------------------------
+
+# BEFORE YOU BEGIN
+# ------------------------------------------------------------------------------
+# 1. Make sure you have the HSM Key plugged in
+# 2. Make sure you remember your HSM PIN
+# 3. To re-test the connection to the HSM, you can do:
```

Comment:
Do you mean re-test or reset?

Reply:
`both` 😅 . `gpgconf --kill all && sudo systemctl restart pcscd` reset the daemon and `pkcs11-tool -O` re-test in case the HSM is not routed to the VM (for example)